### PR TITLE
Optimisations and bug fixes

### PR DESCRIPTION
**Pull request by FungusGenerator**
- Removed some instances of `next` and `pairs`
- Replaced some instances of `wait` with `task.wait` and `delay` with `task.delay`
- Updated BotBrain to replace deprecated objects such as `workspace:FindPartOnRay` and `CFRame.p` with up-to-date ones

- Updated :bot to fix the issue of "walk" always being true and the bot number always being 50 or higher
- Fixed ":command %" being the same as ":command all"

This pull request therefore declares that I have no job
_i should get contributor fr_

### CODE REVIEW

**Comment by P3tray** [comment]
damn son, might as well update the bot command to be better.

File: MainModule/Server/Commands/Moderators.lua [hunk 1]
```diff
@@ -6627,13 +6627,19 @@ return function(Vargs, env)
 			Description = "AI bots made for training; ':bot scel 5 true true'";
 			AdminLevel = "Moderators";
 			Function = function(plr: Player, args: {string})
-				local num = tonumber(args[2]) and math.max(tonumber(args[2]), 50) or 1
+				local num = tonumber(args[2]) and math.min(tonumber(args[2]), 50) or 1
 				local health = tonumber(args[6]) or 100
 				local speed = tonumber(args[7]) or 16
 				local damage = tonumber(args[8]) or 5
-				local walk = args[3] == "false" and false or true
```
Comment (P3tray) [inline MainModule/Server/Commands/Moderators.lua:6634]: revert this change. why'd you make this worse?

File: MainModule/Server/Dependencies/Assets/BotBrain.rbxmx [hunk 2]
```diff
@@ -113,7 +120,7 @@ function validTarget(v)
 	local canHurt = (not isBot and not isPlayer) or (isBot and props.AttackBots and (isBot.Value ~= props.SpecialKey or isBot.Value == "null")) or (isPlayer and not (props.Friendly and (isPlayer == props.Creator or isPlayer.TeamColor == props.TeamColor)))
 	if targ and targ ~= char and canHurt then
 		local targetHumanoid = targ:FindFirstChildOfClass("Humanoid")
-		if targetHumanoid ~= nil and targetHumanoid.Health < 999999999 and (targ:FindFirstChild("HumanoidRootPart") or targ:FindFirstChild("Torso") or targ:FindFirstChild("UpperTorso")) then
+		if targetHumanoid ~= nil and targetHumanoid.Health < 9e9 and (targ:FindFirstChild("HumanoidRootPart") or targ:FindFirstChild("Torso") or targ:FindFirstChild("UpperTorso")) then
```
Comment (P3tray) [inline MainModule/Server/Dependencies/Assets/BotBrain.rbxmx:123]: math.huge exists >:(

**Review by P3tray — CHANGES_REQUESTED** [review]
Revert all your conversions of `task.wait` inside the Adonis env (so not the bot brains file because I'm not entirely sure if it is given access to the Adonis env, probably not). We already overwrite `wait` with `task.wait` at the top of the file.

See other requests also.

Have you tested this properly? Seeing usage of `replace all` gives us Vietnam flashbacks, I know you have used this because you've modified commented out code.

File: MainModule/Server/Commands/Moderators.lua (hunk 1 above)
Comment (P3tray) [reply MainModule/Server/Commands/Moderators.lua:6634]: you'll probably have to expand the code view to see what you've done. there is programattically no reason as to why this should have been done

File: MainModule/Server/Dependencies/Assets/BotBrain.rbxmx (hunk 2 above)
Comment (FungusGenerator) [reply MainModule/Server/Dependencies/Assets/BotBrain.rbxmx]: I remember testing the performance of it, but I guess I didn't do it correctly since testing now they are the same. I replaced 9e9 with math.huge

File: MainModule/Server/Commands/Moderators.lua (hunk 1 above)
Comment (FungusGenerator) [reply MainModule/Server/Commands/Moderators.lua:6634]: Didn't knew `wait` was replaced with `task.wait` by default, I may revert those


If you are talking about line 6634, it always evaluates to true as `x and false or y` will always be y, even Roblox Studio gives an error about it. You can test it if you want, the bots will walk even if you give false

If you are talking about line 6630, I don't see any reason why math.max would be used there. You can also test if this is right, as the bot number would never be lower than 50 if math.max was used


I tested the loader, :bot (with walk as true and false), :alert, :cd, :pchat and :vote, and !ping (I don't know if that matters)

Comment (P3tray) [reply MainModule/Server/Commands/Moderators.lua:6634]: alright

Comment (P3tray) [reply MainModule/Server/Commands/Moderators.lua:6634]: @Expertcoderz head hurty

**Comment by Sceleratis** [comment]
Looks good
